ggshield/cmd/auth/login.py

```diff
@@ -189,7 +190,11 @@ def token_login(config: Config, instance: Optional[str]) -> None:
 
     # enforce using the token (and not use config default)
     client = create_client(api_key=token, api_url=config.api_url)
-    response = client.get(endpoint="token")
+    try:
+        response = client.get(endpoint="token")
+    except requests.exceptions.ConnectionError:
```

Comment:
Hum there could be causes for ConnectionError other than an Invalid instance, where you trying to catch here if the user set an invalid instance ?

Reply:
The problem is the following: it will crash when calling `'$GITGUARDIAN_INSTANCE=https://dashboard.example.com' ggshield auth login` since the instance URL is invalid. Currently, the whole stack trace is displayed in the console, which is not ideal. 
So I would like to catch the error when it happens. The problem is that we can know for sure in advance that the instance URL will be invalid, as it could be an on-prem instance. So I wanted to fix that by catching the error we get when we can't resolve the resource name. 
I agree that this might not be ideal, do you have another idea to handle the error a better way?

Reply:
Hum, when I try on my current ggshield ` GITGUARDIAN_INSTANCE='https://dashboard.example.com' ggshield auth login` does not crash (it just gets stuck because the web page does not identify me), but in deed `GITGUARDIAN_INSTANCE='https://dashboard.example.com' ggshield auth login --method=token` crashes in a problematic way.
The full error is:
```
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='dashboard.example.com', port=443): Max retries exceeded with url: /exposed/v1/token (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x725d5d2a4940>: Failed to resolve 'dashboard.example.com' ([Errno -2] Name or service not known)")"
```
I propose that you modify your solution to detect the  `"Failed to resolve {instance}"` in the error and in this case only raise the invalid instance ? What do you think ?

Reply:
I adjusted the error handling as you suggested, let me know if this works for you